### ISSUE TITLE
Error : opal.js:140  - "Object.defineProperty(object, name,..."

### ISSUE THREAD

**Issue by cmoulliard** [asciidoctor/asciidoctor.js#651]
## Steps to reproduce

```
git clone https://github.com/cmoulliard/litoria.git
nvm use v10.10.0
npm install
npm run test
```

## Error reported

```
> litoria@0.0.32 test /Users/dabou/MyProjects/litoria
> tape test/*.js | tap-spec

/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:140
      Object.defineProperty(object, name, {
             ^

TypeError: Object.defineProperty called on non-object
    at Function.defineProperty (<anonymous>)
    at $defineProperty (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:140:14)
    at Object.Opal.allocate_module (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:546:5)
    at Opal.module (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:593:19)
    at /Users/dabou/MyProjects/litoria/node_modules/asciidoctor-reveal.js/dist/main.js:31:41
    at /Users/dabou/MyProjects/litoria/node_modules/asciidoctor-reveal.js/dist/main.js:36:7
    at Opal.modules.asciidoctor-revealjs/converter (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor-reveal.js/dist/main.js:37:5)
    at Object.Opal.load (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:2300:7)
    at $Object.Opal.require (/Users/dabou/MyProjects/litoria/node_modules/asciidoctor.js/node_modules/opal-runtime/src/opal.js:2328:17)
    at /Users/dabou/MyProjects/litoria/node_modules/asciidoctor-reveal.js/dist/main.js:4978:17
```

**Comment by Mogztter** [comment]
Hello Charles,

It's related to https://github.com/opal/opal/issues/1846
Fundamentals JavaScript objects were altered by Opal. It has been fixed in Opal master, so it should be fixed in the next version of Asciidoctor.js.

As a workaround it's possible to save and restore the function, see https://github.com/Mogztter/opal-node-runtime/blob/3489d53241b1e157ad87264dc3782c697597ed6f/src/index.js#L3

So basically you will have to save `defineProperty` function before requiring Asciidoctor.js (and thus Opal) and restore it after.

Also related: https://github.com/asciidoctor/asciidoctor.js/issues/562  https://github.com/asciidoctor/asciidoctor.js/issues/556

**Comment by Mogztter** [comment]
I can't reproduce your error though, could you please upgrade to the latest version of Asciidoctor.js ?

**Comment by Mogztter** [comment]
```
git clone https://github.com/cmoulliard/litoria.git
nvm use v10.10.0
npm install
npm run test
```
```
  total:     21
  passing:   21
  duration:  856ms
```

**Comment by Mogztter** [comment]
Forget what I said it seems to be an incompatibility between Asciidoctor.js 1.5.9 and Asciidoctor Reveal.js 1.1.3.

**Comment by Mogztter** [comment]
https://github.com/asciidoctor/asciidoctor-reveal.js/issues/227

**Comment by pieh** [comment]
I think we've hit the problem with `opal-runtime` altering fundamental objects ( https://github.com/gatsbyjs/gatsby/issues/13097 ). Seems like forcing `opal-runtime` to use updated version is fixing it (but it require our users to use `yarn` resolutions). 

Any chance you can cut new release that bumps pinned version of `opal-runtime` dependency from `1.0.11` to `1.0.13`?

**Comment by Mogztter** [comment]
Hello @pieh.

> Any chance you can cut new release that bumps pinned version of opal-runtime dependency from 1.0.11 to 1.0.13?

I'm a bit reluctant to do that because the code produces by Opal 1.0.11 and 1.0.13 may not be compatible.
In other words, if I publish a new version of Asciidoctor.js it should be a major version to avoid issues.

Having said that, Asciidoctor.js 2.0.0 is just around the corner. I've already published a second release candidate and hopefully the next will be 2.0.0.

So while it seems to do the trick in the `gatsby-transformer-asciidoc` module I don't want to take the risk to break other libraries that may use Asciidoctor.js with some Ruby code.

**Comment by Mogztter** [comment]
@pieh Asciidoctor 2.0.0 has been released. Could you please confirm that this issue is now resolved ?

**Comment by pieh** [comment]
@Mogztter thanks for heads up. Sorry I didn't notice previous reply, but I will try bumping major version now, see if it solves the issue and report back

**Comment by pieh** [comment]
Can confirm that `ascidoctor@2.0.0` fixes the issue. 

Was bit confused when bumping as `^1` is using `asciidoctor.js` package name and `^2` is using `asciidoctor` (no `.js` suffix) - is the package name change intentional?

**Comment by Mogztter** [comment]
> Can confirm that ascidoctor@2.0.0 fixes the issue.

Awesome 👍 

> Was bit confused when bumping as ^1 is using asciidoctor.js package name and ^2 is using asciidoctor (no .js suffix) - is the package name change intentional?

Yes, we've decided to drop the `.js` suffix because it was unnecessary (https://docs.npmjs.com/files/package.json#name).

In your case I think you can directly use `@asciidoctor/core`. 
`asciidoctor` is a meta package that contains both `@asciidoctor/core` (API) and `@asciidoctor/cli` (CLI).

**Comment by Mogztter** [comment]
> Was bit confused when bumping as ^1 is using asciidoctor.js package name and ^2 is using asciidoctor (no .js suffix) - is the package name change intentional?

I've added a note on the release page but maybe I should also add a note on the README ? What do you think ?

**Comment by pieh** [comment]
> I've added a note on the release page but maybe I should also add a note on the README ? What do you think ?

If You don't intend to use `asciidoctor.js`, I would advise to deprecate that package on npm and point it to new package name - https://docs.npmjs.com/deprecating-and-undeprecating-packages-or-package-versions - this should take care of displaying message on `npm` website.
